### PR TITLE
Support appointment listing and storage in ui

### LINKED ISSUE #85: Create UniquePropertyList class to maintain list of unique properties
Supports addProperty, setProperty, remove property functionality while maintaining a unique property list.

### LINKED ISSUE #86: Create UniqueAppointmentList class to maintain unique list of appointments
Support addAppointment, setAppointment, remove appointment functionality while keeping a unique list of appointments.

### LINKED ISSUE #87: Update PropertyBook and add ReadOnlyPropertyBook class
Replace management of ArrayList<Property>  to UniquePropertyList in PropertyBook.
Create ReadOnlyPropertyBook class to facilitate an unmodifiable view of the property list.

### LINKED ISSUE #88: Update AppointmentBook and add ReadOnlyAppointmentBook class
Replace management of ArrayList<Appointment> to UniqueAppointmentList in AppointmentBook.
Create ReadOnlyAppointmentBook class to facilitate an unmodifiable view of the property list.

### LINKED ISSUE #89: Create Property exception classes to handle invalid Property Commands.
Add DuplicatePropertyException class to handle error throwing related to addition of duplicate properties.
Add PropertyNotFound class to handle error throwing for operational attempts made to properties not found in the list.

### LINKED ISSUE #90: Create Appointment exception classes to handle invalid Appointment Commands.
Add DuplicateAppointmentException class to handle error throwing related to addition of duplicate appointments.
Add AppointmentNotFound class to handle error throwing for operational attempts made to appointments not found in the list.

### LINKED ISSUE #91: Create AppointmentCard, AppointmentList classes and FXML files
Support listing of appointments in Ui.

### LINKED ISSUE #92: Add AppointmentBookStorage, Json classes to support storage of appointments
Add AppointmentBookStorage, JsonAdaptedAppointment, JsonAppointmentBookStorage, JsonSerializableAppointmentBook to support storage of appointment data.

### LINKED ISSUE #93: Update ModelManager, Model classes to support use of AppointmentBook
Remove support for AddressBook and add support for AppointmentBook operations.

### LINKED ISSUE #94: Update MainApp and MainWindow for appointment listing in Ui
Support initialisation of AppointmentBookStorage and display appointments from storage in Ui.

### PR DESCRIPTION
**Pull request by candyhy**
Add UniqueAppointmentList and UniquePropertyList classes to support listing of unique properties and appointments.
Add AppointmentBook, ReadOnlyAppointmentBook, PropertyBook, ReadOnlyPropertyBook classes to manage appointment and property operations.
Add DuplicatePropertyException, PropertyNotFoundException, DuplicateAppointmentException, AppointmentNotFoundException classes to prevent addition of duplicate properties and appointments.
Add AppointmentCard, AppointmentListPanel classes and .fxml files to support listing of appointments in Ui.
Add AppointmentBookStorage, JsonAdaptedAppointment, JsonAppointmentBookStorage, JsonSerializableAppointmentBook to support storage of appointments added.
Update ModelManager, Model classes to support use of AppointmentBook 
Update MainApp and MainWindow to support initialisation of appointment book storage and display appointments from storage.

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedAppointment.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.Date;
+import seedu.address.model.appointment.Time;
+import seedu.address.model.name.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.remark.Remark;
+import seedu.address.model.util.DateTimeFormat;
+
+/**
+ * Jackson-friendly version of {@link Person}.
```
Comment (w2vgd) [inline src/main/java/seedu/address/storage/JsonAdaptedAppointment.java]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -73,6 +73,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case FindCommand.COMMAND_WORD:
             return new FindCommandParser().parse(arguments);
 
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java]: I think this may be removed as it is handled above

**Review by w2vgd — COMMENTED** [review]
LGTM 👍  with some small comments

File: src/main/java/seedu/address/storage/JsonAdaptedAppointment.java (hunk 1 above)
Comment (candyhy) [reply src/main/java/seedu/address/storage/JsonAdaptedAppointment.java]: fixed!

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java (hunk 2 above)
Comment (candyhy) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java]: fixed as well

**Review by w2vgd — APPROVED** [review]
LGTM 👍

**Pull request merged**